### ISSUE TITLE
Travis-CI Cache Problem?

### ISSUE THREAD

**Issue by garyyu** [mimblewimble/grin#1675]
Recently I draft a PR with a build error or with an obvious dead lock problem, but Travis-CI test passed!

I suspect there must be a problem caused by Travis-CI cargo cache, perhaps imported by Travis-CI Speed Optimization PR:  https://github.com/mimblewimble/grin/pull/1647

I have to revert the PR #1647  and try to address the root cause of this Travis-CI problem later.

**Comment by garyyu** [comment]
https://travis-ci.org/mimblewimble/grin/jobs/438144451
```
For more information about this error, try `rustc --explain E0425`.
error: Could not compile `grin_p2p`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```
Looks not like a cache issue, could be the bash script error code acquiring issue.
Anyway, I will give more tests later before re-import the speed optimization.

**Comment by garyyu** [comment]
closed by #1687
